tests/ignite/metrics/test_ndcg.py

```diff
@@ -0,0 +1,266 @@
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-6)
+
+
+def test_reset():
+
+    y_true = torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])
+    y_pred = torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]])
+    ndcg = NDCG()
+    ndcg.update([y_pred, y_true])
+    ndcg.reset()
+
+    with pytest.raises(NotComputableError, match=r"NGCD must have at least one example before it can be computed."):
+        ndcg.compute()
+
+
+def _ndcg_sample_scores(y_true, y_score, k=None, ignore_ties=False):
+
+    gain = _dcg_sample_scores(y_true, y_score, k, ignore_ties=ignore_ties)
+    normalizing_gain = _dcg_sample_scores(y_true, y_true, k, ignore_ties=True)
+    all_irrelevant = normalizing_gain == 0
+    gain[all_irrelevant] = 0
+    gain[~all_irrelevant] /= normalizing_gain[~all_irrelevant]
+    return gain
+
+
+@pytest.mark.parametrize("log_base", [2, 3, 10])
+def test_log_base(log_base):
+    def ndcg_score_with_log_base(y_true, y_score, *, k=None, sample_weight=None, ignore_ties=False, log_base=2):
+
+        gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
+        return np.average(gain, weights=sample_weight)
+
+    y_true = torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]])
+    y_pred = torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])
+
+    ndcg = NDCG(log_base=log_base)
+    ndcg.update([y_pred, y_true])
+
+    result_ignite = ndcg.compute()
+    result_sklearn = ndcg_score_with_log_base(y_true.numpy(), y_pred.numpy(), log_base=log_base)
+
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-6)
+
+
+def test_update(test_case):
+
+    y_pred, y_true = test_case
+
+    y_pred = y_pred
+    y_true = y_true
+
+    y1_pred = torch.multinomial(y_pred, 5, replacement=True)
+    y1_true = torch.multinomial(y_true, 5, replacement=True)
+
+    y2_pred = torch.multinomial(y_pred, 5, replacement=True)
+    y2_true = torch.multinomial(y_true, 5, replacement=True)
+
+    y_pred_combined = torch.cat((y1_pred, y2_pred))
+    y_true_combined = torch.cat((y1_true, y2_true))
+
+    ndcg = NDCG()
+
+    ndcg.update([y1_pred, y1_true])
+    ndcg.update([y2_pred, y2_true])
+
+    result_ignite = ndcg.compute()
+
+    result_sklearn = ndcg_score(y_true_combined.numpy(), y_pred_combined.numpy())
+
+    np.testing.assert_allclose(np.array(result_ignite), result_sklearn, rtol=2e-6)
+
+
+def _test_distrib_output(device):
+
+    rank = idist.get_rank()
+
+    def _test(n_epochs, metric_device):
+
+        metric_device = torch.device(metric_device)
+
+        n_iters = 5
+        batch_size = 8
+        n_items = 5
+
+        torch.manual_seed(12 + rank)
+
+        y_true = torch.rand((n_iters * batch_size, n_items)).to(device)
+        y_preds = torch.rand((n_iters * batch_size, n_items)).to(device)
+
+        def update(_, i):
+            return (
+                [v for v in y_preds[i * batch_size : (i + 1) * batch_size, ...]],
+                [v for v in y_true[i * batch_size : (i + 1) * batch_size]],
+            )
```

Comment:
Why do you return tuple of 2 lists instead of a tuple of two tensors ?

Reply:
I have taken inspiration from the code provided by @puhuk . Here each element of the list is a batch. We feed our engine one batch at a time; hence using a list is also ok. To maintain uniformity across the code, I have kept it this way.

Reply:
Oh, I see, he provided a wrong link. Yes, in accuracy we also a test on list of tensors and numbers but this is untypical. Here is a typical example https://github.com/pytorch/ignite/blob/26f7cec873b382a500204aa0121bff1661248462/tests/ignite/metrics/test_accuracy.py#L412-L463